crypto/mem_sec.c

```diff
@@ -433,8 +437,13 @@ static int sh_init(size_t size, int minsize)
     if (mprotect(sh.map_result + aligned, pgsize, PROT_NONE) < 0)
         ret = 2;
 
+#if defined(MLOCK_ONFAULT) && defined(SYS_mlock2)
```

Comment:
The two defines should be sufficient. However, looking again at the file, the whole thing is wrapped with a Linux-only/Unix only wrapper. Would non-Linux ever define these?